contracts/USM.sol

```diff
@@ -44,52 +57,24 @@ contract USM is ERC20 {
      * @param _usmAmount Amount of USM to burn.
      */
     function burn(uint _usmAmount) external {
-        uint ethAmount = _applyFee(_usmToEth(_usmAmount), BURN_FEE);
-        ethPool = ethPool.sub(ethAmount);
+        uint ethAmount = _usmToEth(_usmAmount);
+        uint ethMinusFee = ethAmount.sub(muld(ethAmount, BURN_FEE));
+        ethPool = ethPool.sub(ethMinusFee);
         _burn(msg.sender, _usmAmount);
-        Address.sendValue(msg.sender, ethAmount);
+        Address.sendValue(msg.sender, ethMinusFee);
     }
 
     /**
      * @notice Calculate debt ratio of the current Eth pool amount and outstanding USM
      * (the amount of USM in total supply).
      *
-     * @return Debt ratio decimal percentage *(10**decimals()). For example, 100%
-     * is denoted as 1,000,000,000,000,000,000 (1 with 18 zeros).
+     * @return Debt ratio in UNIT.
      */
     function debtRatio() external view returns (uint) {
-        return _debtRatio(ethPool, totalSupply());
-    }
-
-    /**
-     * @notice Calculate debt ratio of an Eth pool amount and outstanding USM
-     * (the amount of USM in total supply).
-     *
-     * @param _ethPool Amount of Eth in the pool.
-     * @param _usmOutstanding Amount of USM in total supply.
-     * @return Debt ratio decimal percentage *(10**decimals()). For example, 100%
-     * is denoted as 1,000,000,000,000,000,000 (1 with 18 zeros).
-     */
-    function _debtRatio(uint _ethPool, uint _usmOutstanding) internal view returns (uint) {
-        if (_ethPool == 0) {
+        if (ethPool == 0) {
             return 0;
         }
-        return _usmOutstanding.mul(10**uint(decimals())).div(_ethToUsm(_ethPool));
-    }
-
-    /**
-     * @notice Apply fees to a number, the result being the number, minus the
-     * percentage fee.
-     *
-     * @param _number The amount that the fee is being applied to
-     * @param _feePercentage Percentage fee to be subtracted from the number
-     * in parts per 10,000.
-     * @return The result of the _number minus the percentage fee.
-     */
-    function _applyFee(uint _number, uint _feePercentage) internal pure returns (uint) {
-        uint result = _number.sub((_number.mul(_feePercentage)).div(PERCENTAGE_BASE));
-        require(result < _number, "Fee must make the result smaller");
-        return result;
+        return divd(totalSupply().mul(UNIT), _ethToUsm(ethPool)); // This means a pool upper limit of about 10**50 wei.
```

Comment:
`debtRatio` is reduced to a one-liner. Since both operands are `integer` and we want the result to be `decimal`, we multiply the first one by `UNIT`

Reply:
I think I'm understanding this right... If I am it changes the current `debtRatio` from returning a below `1` value when overcollateralized to a below `100` value. For example, what was 0.9, is now 90 (as in 90%)?

Using smaller numbers (without the constraints of solidity in this thought example), let's assume the following:

* `totalSupply` in USM = 100
* `_ethToUsm(ethPool)` in USM = 110

.. technically we are slightly overcollateralized for whatever reason, which is where we want to be.

Let's assume `UNIT` = 10 (purely for this thought example).

Running this function would perform the following:

1. `totalSupply().mul(UNIT)` = 100 * 10 = 1,000
2. `x.mul(UNIT).div(y)` = (1,000 * 10) / 110 = 10,000 / 110 = 90.909090....

~90.91% debt ratio, which looks right!

I believe that in the master branch, the result would be ~0.909.

Am I getting that right?

Reply:
Before `debtRatio` was returning a fixed point number with `decimals()` decimals, now it is returning a fixed point number with 27 decimals, which is the format for all fixed point numbers in the contract.

I think that to fix the tests, I just need to add 9 zeros to each one of your `debtRatio` tests.